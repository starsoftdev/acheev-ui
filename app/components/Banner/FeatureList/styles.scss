@import '~styles/settings/settings';

.bannerFeatureList {
  &__title {
    font-size: rem-calc(36);
    font-weight: bold;
    line-height: $rem-xxl;
    text-align: center;
    margin-bottom: $rem-md;
    @include breakpoint(small only) {
      font-size: $fs-xxl;
      line-height: $fs-hg;
    }
  }
  &__subTitle {
    font-size: $fs-xl;
    line-height: $rem-lg;
    text-align: center;
    margin-bottom: $rem-mx;
    @include breakpoint(small only) {
      font-size: $fs-mx;
      line-height: $fs-xl;
    }
  }
}
