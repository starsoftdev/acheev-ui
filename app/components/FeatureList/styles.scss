@import '~styles/settings/settings';

.featureList {
  &__item {
    display: flex;
    flex-direction: column;
    align-items: center;
    text-decoration: none;
    color: $body-font-color;
  }
  &__content {
    font-size: $fs-mx;
    font-weight: bold;
    text-transform: uppercase;
    text-align: center;
  }
}
