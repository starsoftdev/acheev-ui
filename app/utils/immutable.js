// @flow

import { Iterable } from 'immutable';
import type { Map } from 'immutable';
import { get as _get } from 'lodash-es';

export const get = (
  collection: ?Map<*, *> | Object,
  path: string | number,
  defaultValue: any
) => {
  if (!collection) {
    return defaultValue;
  } else if (Iterable.isIterable(collection)) {
    return collection.get(path, defaultValue);
  }
  return _get(collection, path, defaultValue);
};

export const getIn = (
  collection: ?Map<*, *> | Object,
  path: Array<string | number>,
  defaultValue: any
) => {
  if (!collection) {
    return defaultValue;
  } else if (Iterable.isIterable(collection)) {
    return collection.getIn(path, defaultValue);
  }
  return _get(collection, path, defaultValue);
};
