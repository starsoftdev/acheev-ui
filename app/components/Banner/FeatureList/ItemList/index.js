// @flow

import * as React from 'react';
import cx from 'classnames';
import { generate } from 'shortid';
import type { List, Map } from 'immutable';

import Icon from 'components/Icon';

import './styles.scss';

type Props = {
  data: List<Map<*, *>>,
  className?: string,
};

const FeatureList = ({ data, className }: Props) => {
  const mergedClassName = cx('itemList row align-spaced', className);
  return (
    <div className={mergedClassName}>
      {data.map(item => (
        <div
          className="itemList__item column medium-4 small-12 mb-mx"
          key={generate()}
        >
          <Icon glyph={item.get('icon', '')} size={70} className="mb-sm" />
          <div className="itemList__title">{item.get('title', '')}</div>
          <div className="itemList__subTitle">{item.get('subTitle', '')}</div>
        </div>
      ))}
    </div>
  );
};

export default FeatureList;
