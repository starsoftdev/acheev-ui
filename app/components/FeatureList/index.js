// @flow

import * as React from 'react';
import cx from 'classnames';
import { generate } from 'shortid';
import Link from 'components/Link';
import { fromJS } from 'immutable';
import type { List, Map } from 'immutable';

import Icon from 'components/Icon';
import DoctorIcon from 'images/sprite/doctor.svg';
import ProducerIcon from 'images/sprite/producer.svg';
import ReviewIcon from 'images/sprite/review.svg';
import StrainIcon from 'images/sprite/strain.svg';

import './styles.scss';

type Props = {
  data?: List<Map<*, *>>,
  className?: string,
};

const defaultFeatureList = fromJS([
  {
    icon: DoctorIcon,
    content: 'Find a clinic',
    linkTo: 'clinics',
  },
  {
    icon: ProducerIcon,
    content: 'find a producer',
    linkTo: 'producers',
  },
  {
    icon: StrainIcon,
    content: 'find a strain',
    linkTo: 'strains',
  },
  {
    icon: ReviewIcon,
    content: 'Leave reviews & earn Points',
    linkTo: 'rewards',
  },
]);

const FeatureList = ({ data = defaultFeatureList, className }: Props) => {
  const mergedClassName = cx('featureList row align-spaced', className);
  return (
    <div className={mergedClassName}>
      {data.map(item => (
        <Link
          className={cx('featureList__item column medium-3 mb-md', {
            'small-4': item.get('linkTo') !== 'rewards',
            'small-12': item.get('linkTo') === 'rewards',
          })}
          to={item.get('linkTo', '')}
          key={generate()}
        >
          <Icon
            glyph={item.get('icon', '')}
            width={80}
            height={80}
            className="mb-mx"
          />
          <div className="featureList__content">{item.get('content', '')}</div>
        </Link>
      ))}
    </div>
  );
};

export default FeatureList;
