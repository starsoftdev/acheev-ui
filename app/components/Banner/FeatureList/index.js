// @flow

import * as React from 'react';

import { fromJS } from 'immutable';

import UserIcon from 'images/sprite/user.svg';
import RecommendIcon from 'images/sprite/recommend.svg';
import ReviewsIcon from 'images/sprite/product-reviews.svg';

import ItemList from './ItemList';
import './styles.scss';

const data = fromJS([
  {
    icon: UserIcon,
    title: '1,000,000',
    subTitle: 'Annual users',
  },
  {
    icon: ReviewsIcon,
    title: '60,000',
    subTitle: 'Product reviews',
  },
  {
    icon: RecommendIcon,
    title: '9/10 Users',
    subTitle: 'Recommend Lift & Co.',
  },
]);

const BannerFeatureList = () => (
  <div className="bannerFeatureList row mb-lg">
    <div className="small-12 large-12 column">
      <h2 className="bannerFeatureList__title">
        Search smarter. Consume confidently.
      </h2>
    </div>
    <div className="small-12 large-8 large-offset-2 columm">
      <div className="bannerFeatureList__subTitle">
        We&#39;ve created an algorithm that maps millions of user and product
        data points to deliver personalized cannabis recommendations.
      </div>
    </div>

    <div className="small-12 large-12 columm">
      <ItemList data={data} />
    </div>
  </div>
);

export default BannerFeatureList;
