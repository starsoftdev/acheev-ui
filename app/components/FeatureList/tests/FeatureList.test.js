import React from 'react';
import { mount } from 'enzyme';
import { fromJS } from 'immutable';
import { create } from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import FeatureList from 'components/FeatureList';

import ProducerIcon from 'images/sprite/producer.svg';
import ReviewIcon from 'images/sprite/review.svg';
import StrainIcon from 'images/sprite/strain.svg';

const data = fromJS([
  {
    icon: StrainIcon,
    content: 'find a strain',
    linkTo: 'strains',
  },
  {
    icon: ProducerIcon,
    content: 'find a producer',
    linkTo: 'producers',
  },
  {
    icon: ReviewIcon,
    content: 'Leave reviews & earn Points',
    linkTo: 'rewards',
  },
]);

describe('<FeatureList />', () => {
  it('should apply className prop', () => {
    const renderedComponent = mount(
      <MemoryRouter>
        <FeatureList className="hey" />
      </MemoryRouter>
    );
    expect(renderedComponent.find('.featureList').hasClass('hey')).toEqual(
      true
    );
  });

  it('renders default feature list if data is not provided', () => {
    const component = create(
      <MemoryRouter>
        <FeatureList />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders data if provided', () => {
    const component = create(
      <MemoryRouter>
        <FeatureList data={data} />
      </MemoryRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
