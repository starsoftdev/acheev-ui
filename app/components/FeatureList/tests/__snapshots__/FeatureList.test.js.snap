// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FeatureList /> renders data if provided 1`] = `
<div
  className="featureList row align-spaced"
>
  <a
    className="featureList__item column medium-3 mb-md small-4"
    href="/strains"
    onClick={[Function]}
  >
    <svg
      className="mb-mx"
      height={80}
      width={80}
    >
      <use
        xlinkHref="#undefined"
      />
    </svg>
    <div
      className="featureList__content"
    >
      find a strain
    </div>
  </a>
  <a
    className="featureList__item column medium-3 mb-md small-4"
    href="/producers"
    onClick={[Function]}
  >
    <svg
      className="mb-mx"
      height={80}
      width={80}
    >
      <use
        xlinkHref="#undefined"
      />
    </svg>
    <div
      className="featureList__content"
    >
      find a producer
    </div>
  </a>
  <a
    className="featureList__item column medium-3 mb-md small-12"
    href="/rewards"
    onClick={[Function]}
  >
    <svg
      className="mb-mx"
      height={80}
      width={80}
    >
      <use
        xlinkHref="#undefined"
      />
    </svg>
    <div
      className="featureList__content"
    >
      Leave reviews & earn Points
    </div>
  </a>
</div>
`;

exports[`<FeatureList /> renders default feature list if data is not provided 1`] = `
<div
  className="featureList row align-spaced"
>
  <a
    className="featureList__item column medium-3 mb-md small-4"
    href="/clinics"
    onClick={[Function]}
  >
    <svg
      className="mb-mx"
      height={80}
      width={80}
    >
      <use
        xlinkHref="#undefined"
      />
    </svg>
    <div
      className="featureList__content"
    >
      Find a clinic
    </div>
  </a>
  <a
    className="featureList__item column medium-3 mb-md small-4"
    href="/producers"
    onClick={[Function]}
  >
    <svg
      className="mb-mx"
      height={80}
      width={80}
    >
      <use
        xlinkHref="#undefined"
      />
    </svg>
    <div
      className="featureList__content"
    >
      find a producer
    </div>
  </a>
  <a
    className="featureList__item column medium-3 mb-md small-4"
    href="/strains"
    onClick={[Function]}
  >
    <svg
      className="mb-mx"
      height={80}
      width={80}
    >
      <use
        xlinkHref="#undefined"
      />
    </svg>
    <div
      className="featureList__content"
    >
      find a strain
    </div>
  </a>
  <a
    className="featureList__item column medium-3 mb-md small-12"
    href="/rewards"
    onClick={[Function]}
  >
    <svg
      className="mb-mx"
      height={80}
      width={80}
    >
      <use
        xlinkHref="#undefined"
      />
    </svg>
    <div
      className="featureList__content"
    >
      Leave reviews & earn Points
    </div>
  </a>
</div>
`;
